lbry/db/queries/txio.py

```diff
@@ -143,28 +143,35 @@ def distribute_unspent_txos(
     return total, buckets
 
 
-def where_changed_support_txos(blocks: Optional[Tuple[int, int]]):
+def claims_with_changed_supports(blocks: Optional[Tuple[int, int]]) -> Select:
+    has_changed_supports = (
+        select(Claim.c.claim_hash.label("claim_hash"), Claim.c.channel_hash.label("channel_hash"))
+        .join(Claim, Claim.c.claim_hash == TXO.c.claim_hash)
+        .where(
+            (TXO.c.txo_type == TXO_TYPES['support']) &
+            (between(TXO.c.height, blocks[0], blocks[-1]) | between(TXO.c.spent_height, blocks[0], blocks[-1]))
+        )
+        .cte("has_changed_supports")
+    )
+
     return (
-        (TXO.c.txo_type == TXO_TYPES['support']) & (
-            between(TXO.c.height, blocks[0], blocks[-1]) |
-            between(TXO.c.spent_height, blocks[0], blocks[-1])
+        select(has_changed_supports.c.claim_hash)
+        .union_all(  # UNION ALL is faster than UNION because it does not remove duplicates
+            select(has_changed_supports.c.channel_hash)
+            .where(has_changed_supports.c.channel_hash.isnot(None))
         )
     )
 
 
-def where_claims_with_changed_supports(blocks: Optional[Tuple[int, int]]):
+def where_claims_with_changed_supports(blocks: Optional[Tuple[int, int]]) -> Select:
     return Claim.c.claim_hash.in_(
-        select(TXO.c.claim_hash).where(
-            where_changed_support_txos(blocks)
-        )
+        claims_with_changed_supports(blocks)
     )
 
 
 def count_claims_with_changed_supports(blocks: Optional[Tuple[int, int]]) -> int:
-    sql = (
-        select(func.count(distinct(TXO.c.claim_hash)).label('total'))
-        .where(where_changed_support_txos(blocks))
-    )
+    subq = claims_with_changed_supports(blocks).subquery()
```

Comment:
@eukreign i think this pattern would work better for us going forward. instead of passing the aggregate to the function to select the right thing (e.g. `staked_support_count_calc`), have the function return a query and use that as a subquery to `count(distinct())`

Reply:
if possible, could you rename or spell out the variable `subq` in full to stay with the general Python naming guidance of using whole words